### PR TITLE
azurerm_bot_channels_registration - support for cmk_key_vault_url, description, icon_url, isolated_network_enabled

### PR DESCRIPTION
**Pull request by neil-yechenwei**
This PR is introducing new properties for azurerm_bot_channels_registration.

image

API Reference:
https://github.com/Azure/azure-rest-api-specs/blob/2a8a9a79b23c72c6092d5b1212acf6d69b1b3850/specification/botservice/resource-manager/Microsoft.BotService/stable/2021-03-01/botservice.json#L36

### CODE REVIEW

File: website/docs/r/bot_channels_registration.markdown [hunk 1]
```diff
@@ -43,8 +43,16 @@ The following arguments are supported:
 
 * `microsoft_app_id` - (Required) The Microsoft Application ID for the Bot Channels Registration. Changing this forces a new resource to be created.
 
+* `cmek_key_vault_url` - (Optional) The CMEK Key Vault Key URL to encrypt the Bot Channels Registration with the Customer Managed Encryption Key.
```
Comment (katbyte) [inline website/docs/r/bot_channels_registration.markdown:46]: given that every other resource uses cmk
```suggestion
* `cmk_key_vault_url` - (Optional) The CMEK Key Vault Key URL to encrypt the Bot Channels Registration with the Customer Managed Encryption Key.
```

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/bot_channels_registration.markdown (hunk 1 above)
Comment (neil-yechenwei) [reply website/docs/r/bot_channels_registration.markdown]: Updated

**Comment by neil-yechenwei** [comment]
@katbyte , thanks for your comments. I've updated code. Please have another look. Thanks.

File: azurerm/internal/services/bot/bot_channels_registration_resource.go [hunk 2]
```diff
@@ -124,6 +138,18 @@ func resourceBotChannelsRegistration() *pluginsdk.Resource {
...
 			},
 
+			"icon_url": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validate.BotChannelRegistrationIconUrl,
+			},
+
+			"is_isolated_network_enabled": {
```
Comment (katbyte) [inline azurerm/internal/services/bot/bot_channels_registration_resource.go:148]: we can probably remove is?
```suggestion
			"isolated_network_enabled": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]

File: azurerm/internal/services/bot/bot_channels_registration_resource.go (hunk 2 above)
Comment (neil-yechenwei) [reply azurerm/internal/services/bot/bot_channels_registration_resource.go]: Updated

**Comment by neil-yechenwei** [comment]
@katbyte , thanks for your comments. I've updated code. Please have another look. Thanks.

**Review by katbyte — APPROVED** [review]
Thanks @neil-yechenwei - LGTM 🚀

**Pull request merged**